Modules/Devices/src/Devices.Application/Identities/Commands/ApproveDeletionProcess/Handler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Application.IntegrationEvents.Outgoing;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using MediatR;
+
+namespace Backbone.Modules.Devices.Application.Identities.Commands.ApproveDeletionProcess;
+
+public class Handler : IRequestHandler<ApproveDeletionProcessCommand, ApproveDeletionProcessResponse>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly IUserContext _userContext;
+    private readonly IEventBus _eventBus;
+
+    public Handler(IIdentitiesRepository identitiesRepository, IUserContext userContext, IEventBus eventBus)
+    {
+        _identitiesRepository = identitiesRepository;
+        _userContext = userContext;
+        _eventBus = eventBus;
+    }
+
+    public async Task<ApproveDeletionProcessResponse> Handle(ApproveDeletionProcessCommand request, CancellationToken cancellationToken)
+    {
+        var identity = await _identitiesRepository.FindByAddress(_userContext.GetAddress(), cancellationToken) ?? throw new NotFoundException(nameof(Identity));
+
+        var deletionProcess = identity.ApproveDeletionProcess(_userContext.GetDeviceId(), request.DeletionProcessId);
```

Comment:
I've tried passing an `DeletionProcessId` object to the `Identity.ApproveDeletionProcess` instead of a string, the problem is that the validation of that value is done in the `IdentityDeletionProcessId.Create` method. If that validation does not pass (which is the one of the test scenario cases) there is no `Value` to be retrieved and thus a different exception is thrown than the one expected.

The test scenario case mentioned is `Modules.Devices.Application.Tests.Tests.Identities.Commands.ApproveDeletionProcess.Throws_when_deletion_process_does_not_exist`. A workaround would be to check if the validation of the string `someNonExistentDeletionProcessId` passes, and inspecting the returning exception if not. 

As the test is currently configured, I am afraid it would only work with string as input to the `ApproveDeletionProcess` method.

Reply:
And what's the problem? Why don't you simply use a different id in the test?

Reply:
(one that is valid and therefore can be parsed)

Reply:
Did just that.